apps/bfd-data-fda/Jenkinsfile

```diff
@@ -0,0 +1,138 @@
+ --domain-owner "$ACCOUNT_ID" \
+ --output text --query authorizationToken
+'''
+                            ).trim()
+                            codeArtifactEndpoint = sh(
+                                returnStdout: true,
+                                script: '''
+aws codeartifact get-repository-endpoint \
+--domain bfd-mgmt --repository bfd-mgmt \
+--format maven --output text
+'''
+                            )
+                            env.CODEARTIFACT_AUTH_TOKEN = codeArtifactAuthToken
+                            env.CODEARTIFACT_ENDPOINT = codeArtifactEndpoint
+                        }
+                        sh '''
+cat <<EOF > ~/.m2/settings.xml
+<settings xmlns="http://maven.apache.org/settings/1.0.0" xmlns:xsi="http://www.w3.org/2001/xmlschema-instance"
+xsi:schemalocation="http://maven.apache.org/settings/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd">
+<servers>
+    <server>
+    <username>aws</username>
+    <password>${CODEARTIFACT_AUTH_TOKEN}</password>
+    <id>bfd-mgmt-bfd-mgmt</id>
+    </server>
+</servers>
+</settings>
+EOF
+'''
+                    }
+                }
+            }
+        }
+        stage('Maven Build') {
+            steps {
+                script {
+                    container('bfd-cbc-build') {
+                    // Calculate FDA DrugCode Sha256
+                        fdaDrugCodeSha256 = sh(
+                            returnStdout: true,
+                            script: '''
+curl -fsSL https://www.accessdata.fda.gov/cder/ndctext.zip | sha256sum | awk '{ print $1 }'
+'''
+                        )
+                        awsAuth.assumeRole()
+                        ssmLookupSha256 = sh(
+                            returnStdout: true,
+                            script: '''
+if parameter_lookup="$(aws ssm get-parameter --name /bfd/mgmt/versioning/nonsensitive/bfd-data-fda \
+--region us-east-1 --output text --query Parameter.Value 2>/dev/null)"; then
+   echo "$parameter_lookup"
+fi
+'''
+                        )
+
+                        if (ssmLookupSha256.isEmpty() || fdaDrugCodeSha256.trim() != ssmLookupSha256.trim()) {
+                            dir('apps/bfd-data-fda') {
+                                def quietFlags = params.verbose_mvn_logging ? '' : '--quiet'
+                                sh "mvn ${quietFlags} clean install"
+                                sh """
+mvn ${quietFlags} exec:java \
+-Dexec.mainClass="gov.cms.bfd.data.fda.utility.App" \
+-Dexec.args="./src/main/resources/"
+"""
+                                sh 'echo "$CODEARTIFACT_ENDPOINT"'
+                                sh "mvn ${quietFlags} package"
+
+                                withCredentials([string(credentialsId: 'bfd-aws-account-id', variable: 'ACCOUNT_ID')]) {
+                                    sh '''
+mvn deploy "-DaltDeploymentRepository=bfd-mgmt-bfd-mgmt::default::https://bfd-mgmt-${ACCOUNT_ID}.d.codeartifact.us-east-1.amazonaws.com/maven/bfd-mgmt/"
```

Comment:
A note here about using this style of templated string instead of re-using `$CODEARTIFACT_ENDPOINT` might be helpful. Baking the domain/repository into this string is a little more brittle. I'm still a little puzzled why the equivalent seemed to yield:

> Invalid syntax for alternative repository. Use "id::layout::url".

Reply:
Revisiting this... it might be fine as it is, but leaving _something_ behind for the next engineer to understand why we did _this_ instead of _that_ might be helpful, limited to a single line admitting that we just didn't go much further in trying to understand why the Jenkins didn't accept the cli-provided `id::layout::url` format with one interpolated value over another would be sufficient, but I'll leave that up to you.